### PR TITLE
Added clear all button

### LINKED ISSUE #6365: [Feature Request] Add clear all button in the Ongoing tasks flyout
### What's the Problem?

Users have to click on each task to clear it from the Ongoing tasks flyout. This is an annoyance when someone's doing a lot of file operations. 

### Solution/Idea

I want to have the *Ongoing tasks* flyout have a Clear all button to clear all inactive / currently not progressing tasks. 

This is how it looks like currently: 
image

This is how I want it to look like: 
image

The highlighted area will have a button button "Clear all" to clear all inactive tasks in one click.

### Alternatives

Nothing so far

### Priorities

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow end users to clear all inactive tasks in one click | Must |
| This button could be a text button saying "Clear all" instead of an X button | Could |

### Files Version

2.0.16.0 

### Windows Version

Windows 11 Pro 21H2 22000.194 in production 

### Additional comment

_No response_

### PR DESCRIPTION
**Pull request by petrjancar**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #6365 

**Details of Changes**
Add details of changes here.
- Added button  "Dismiss all" to clear all inactive tasks in one click.

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@petrjancar can you upload a screenshot of the changes?

**Comment by petrjancar** [comment]
Screenshot 2021-11-22 233109

**Comment by petrjancar** [comment]
@yair100
Screenshot 2021-11-23 000356

**Comment by yaichenbaum** [comment]
@petrjancar very nice! Does it cancel ongoing operations as well?

File: Files/UserControls/OngoingTasksFlyout.xaml [hunk 1]
```diff
@@ -127,6 +127,21 @@
...
+                AutomationProperties.Name="Dismiss All"
+                Background="Transparent"
+                BorderBrush="Transparent"
+                Click="DismissAllBannersButton_Click"
+                ToolTipService.ToolTip="Dismiss All" 
+                HorizontalAlignment="Right">
+                <Button.Content>
+                    <TextBlock Text="Dismiss All" />
+                </Button.Content>
+            </Button>
```
Comment (yaichenbaum) [inline Files/UserControls/OngoingTasksFlyout.xaml:144]:
```suggestion
            <Button
                x:Name="DismissAllBannersButton"                       
                Height="32"                                
                Text="{helpers:ResourceString Name=ClearAll}"
                Background="Transparent"
                BorderBrush="Transparent"
                Click="DismissAllBannersButton_Click"
                ToolTipService.ToolTip="Dismiss All" 
                HorizontalAlignment="Right"/>

Comment (yaichenbaum) [reply Files/UserControls/OngoingTasksFlyout.xaml:144]: Can you add a string called "ClearAll" to the en-us resw file?

**Review by yair100 — CHANGES_REQUESTED** [review]

**Comment by petrjancar** [comment]
> @petrjancar very nice! Does it cancel ongoing operations as well?

No, it only clears all inactive tasks, as requested.

File: Files/UserControls/OngoingTasksFlyout.xaml (hunk 1 above)
Comment (petrjancar) [reply Files/UserControls/OngoingTasksFlyout.xaml]: > Can you add a string called "ClearAll" to the en-us resw file?

Sure, done.

**Comment by yaichenbaum** [comment]
@petrjancar that might be confusing if there are other ongoing operations, do you have any thoughts on how to improve the UX?

**Comment by petrjancar** [comment]
> "Dismiss completed"?

Sounds good to me. What do you think @yair100?

**Comment by yaichenbaum** [comment]
@petrjancar looking at the draft of the design spec, it looks like there are plans to have a separate section on the bottom for completed tasks so in the end "clear all" should work.
image

**Comment by petrjancar** [comment]
> @petrjancar looking at the draft of the design spec, it looks like there are plans to have a separate section on the bottom for completed tasks at the bottom of the flyout so it turns out that "clear all" should work. image

Ok, I updated string in en-us .resw file to "Clear all" instead of "Dismiss all".

**Comment by yaichenbaum** [comment]
@petrjancar in another PR, do you want to create a new section to seperate ongoing and completed tasks?

File: Files/UserControls/OngoingTasksFlyout.xaml [hunk 2]
```diff
@@ -127,6 +127,19 @@
...
+                Height="32"                                
+                Background="Transparent"
+                BorderBrush="Transparent"
+                Click="DismissAllBannersButton_Click"
+                ToolTipService.ToolTip="{helpers:ResourceString Name=ClearAll}" 
+                HorizontalAlignment="Right">
+                <Button.Content>
+                    <TextBlock Text="{helpers:ResourceString Name=ClearAll}" />
+                </Button.Content>
+            </Button>
```
Comment (yaichenbaum) [inline Files/UserControls/OngoingTasksFlyout.xaml:142]:
```suggestion
            <Button
                x:Name="DismissAllBannersButton"                             
                Height="32"                                
                Background="Transparent"
                BorderBrush="Transparent"
                Click="DismissAllBannersButton_Click"
                Content="{helpers:ResourceString Name=ClearAll}"
                ToolTipService.ToolTip="{helpers:ResourceString Name=ClearAll}" 
                HorizontalAlignment="Right"/>
```

**Comment by petrjancar** [comment]
> @petrjancar in another PR, do you want to create a new section to seperate ongoing and completed tasks?

Thanks but not now, I am currently working on other PR.

**Review by yair100 — APPROVED** [review]

**Review by d2dyno1 — APPROVED** [review]

**Comment by yaichenbaum** [comment]
@petrjancar thank you!

**Pull request merged**